src/index.js

```diff
@@ -0,0 +1,164 @@
+const path = require('path');
+const escapeHtml = require('escape-html');
+const pluralize = require('pluralize');
+const helpers = require('atom-linter');
+
+const COMMAND_CONFIG_KEY = 'linter-rubocop.command';
+const DISABLE_CONFIG_KEY = 'linter-rubocop.disableWhenNoConfigFile';
+const OLD_EXEC_PATH_CONFIG_KEY = 'linter-rubocop.executablePath';
+const OLD_ARGS_CONFIG_KEY = 'linter-rubocop.additionalArguments';
+const DEFAULT_LOCATION = { line: 1, column: 1, length: 0 };
+const DEFAULT_ARGS = [
+  '--cache', 'false',
+  '--force-exclusion',
+  '--format', 'json',
+  '--display-style-guide',
+];
+const AUTOCORRECT_ARG = '--auto-correct';
+const STD_IN_ARG = '--stdin';
+
+const DEFAULT_MESSAGE = 'Unknown Error';
+const WARNINGS = new Set(['refactor', 'convention', 'warning']);
+
+const convertOldConfig = () => {
+  const execPath = atom.config.get(OLD_EXEC_PATH_CONFIG_KEY);
+  const args = atom.config.get(OLD_ARGS_CONFIG_KEY);
+  if (!execPath && !args) { return null; }
+  atom.config.set(COMMAND_CONFIG_KEY, `${execPath || ''} ${args || ''}`.trim());
+  atom.config.set(OLD_EXEC_PATH_CONFIG_KEY, undefined);
+  return atom.config.set(OLD_ARGS_CONFIG_KEY, undefined);
+};
+
+// TODO: figure out how to best handle, also, ESLint telling me conflicting things
+// eslint-disable-next-line no-confusing-arrow
+const guard = (value, transform) => (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
+
+const extractUrl = (parsedMessage) => {
+  const [message, url] = Array.from(parsedMessage.split(/ \((.*)\)/, 2));
+  return { message, url };
+};
+
+const formatMessage = ({ message, cop_name: copName, url }) => {
+  const formattedMessage = escapeHtml(message || DEFAULT_MESSAGE);
+  let formattedCopName = '';
+  if (copName != null && url != null) {
+    formattedCopName = ` (<a href="${escapeHtml(url)}">${escapeHtml(copName)}</a>)`;
+  } else if (copName != null) {
+    formattedCopName = ` (${escapeHtml(copName)})`;
+  }
+  return formattedMessage + formattedCopName;
+};
+
+const parseFromStd = (stdout, stderr) => {
+  let parsed;
+  try {
+    parsed = JSON.parse(stdout);
+  } catch (error) {
+    // continue regardless of error
+  }
+  if (typeof parsed !== 'object') { throw new Error(stderr || stdout); }
+  return parsed;
+};
+
+const forwardRubocopToLinter =
+  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
+    const { message, url } = extractUrl(rawMessage);
+    const { line, column, length } = location || DEFAULT_LOCATION;
```

Comment:
If you do `helpers.generateRange(textEditor, 0)` you will get a range highlighting the entire first line. As it stands you would actually be better not specifying a range at all since the current `DEFAULT_LOCATION` will result in a range of 0 characters, which leads to it being invisible in the editor... but still there.